raster-foundry/raster-foundry
# Project embed UI

## Overview

This PR adds the embed page to the project settings area.

The user can select layers from the left side and then choose from the base layer or the analyses based on that layer to add to the embed. The embed URL is not live. If the`EMBED_URI` property in the `global.js` or the overrides file is an empty string, the embed page is not visible.

The parameters expected are the following:
```
projectID=<PROJECTID>
        &mapToken=<MAPTOKEN>
        &type=comparison || single
        &pane1=<LAYERID>
        &pane2=<LAYERID>
        &layers=<LAYERID>,<LAYERID>,<LAYERID>...
        &analyses=<ANALYSISID>,<ANALYSISID>,<ANALYSISID>...
```

* `pane2` should be empty if type is `single`
* `mapToken` should only be present for non-public projects

This PR also fixes a bug in the project service that make the publishing page work again (which is needed to switch the project from public to private within the v2 UI).

### Checklist


### Demo

![image](https://user-images.githubusercontent.com/2442245/54799895-8df2fb80-4c35-11e9-96af-5f7016c0c557.png)

![image](https://user-images.githubusercontent.com/2442245/54799902-93e8dc80-4c35-11e9-9a6e-214622f3b084.png)

![image](https://user-images.githubusercontent.com/2442245/54799912-9d724480-4c35-11e9-8933-a58556def1d6.png)

![image](https://user-images.githubusercontent.com/2442245/54799926-acf18d80-4c35-11e9-92ac-44294fcd7596.png)


### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.


## Testing Instructions
 * Keeping the `global.js` entry for `EMBED_URI` as an empty string, check that you cannot see the Embed page under project settings in the v2 UI
 * Update `EMBED_URI` in your overrides or in your `global.js` file and restart the dev server
 * Ensure that you can see the embed page
 * Notice that the project publishing page now works again
 * On the embed page, add different layers and analyses and check that the url parameters make sense
 * Make the project public/private and see that the map-token is included as needed
 * Try selecting a layer with more than 10 analyses and try loading more analyses

Closes #4744
